src/pmi/src/pmi_v1.c

```diff
@@ -1215,7 +1215,9 @@ static int PMIi_InitIfSingleton(void)
 
     /* FIXME: We need to support a distinct kvsname for each
      * process group */
-    PMI_KVS_Put(singinit_kvsname, cached_singinit_key, cached_singinit_val);
+    if (cached_singinit_inuse) {
```

Comment:
Could you add a comment -- "only push the cached kvs if we have any"?  It is not obvious what `if (cached_singinit_inuse)` means.

Reply:
done